From 7244f5288a64d79af82996c52793d83b07cd5dbf Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Tue, 22 Nov 2011 18:23:01 +0200
Subject: [PATCH 0629/1092] am335x-evm: beta evm bringup changes for wlan and
 bt

* Fix a wrong gpio pin used for bluetooth enable on the beta evm.
  was using gpio0_17 while it should be gpio3_21

* Add a 70 msec delay after turning wlan on to allow for all voltages
  to settle down before starting sdio activity

Signed-off-by: Eyal Reizer <eyalr@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 8 +++++---
 arch/arm/mach-omap2/mux33xx.c         | 2 +-
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index c2d7775..f33f8c7 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -815,7 +815,7 @@ static struct pinmux_config uart1_wl12xx_pin_mux[] = {
 static struct pinmux_config wl12xx_pin_mux_evm_rev1_1a[] = {
 	{"gpmc_a0.gpio1_16", OMAP_MUX_MODE7 | AM33XX_PIN_OUTPUT},
 	{"mcasp0_ahclkr.gpio3_17", OMAP_MUX_MODE7 | AM33XX_PIN_INPUT},
-	{"mcasp0_ahclkx.gpio0_17", OMAP_MUX_MODE7 | AM33XX_PIN_OUTPUT},
+	{"mcasp0_ahclkx.gpio3_21", OMAP_MUX_MODE7 | AM33XX_PIN_OUTPUT},
 	{NULL, 0},
  };
 
@@ -1134,8 +1134,10 @@ static void wl12xx_bluetooth_enable(void)
 
 static int wl12xx_set_power(struct device *dev, int slot, int on, int vdd)
 {
-	if (on)
+	if (on) {
 		gpio_set_value(am335xevm_wlan_data.wlan_enable_gpio, 1);
+		mdelay(70);
+	}
 	else
 		gpio_set_value(am335xevm_wlan_data.wlan_enable_gpio, 0);
 
@@ -1151,7 +1153,7 @@ static void wl12xx_init(int evm_id, int profile)
 	/* Register WLAN and BT enable pins based on the evm board revision */
 	if (gp_evm_revision == GP_EVM_REV_IS_1_1A) {
 		am335xevm_wlan_data.wlan_enable_gpio = GPIO_TO_PIN(1, 16);
-		am335xevm_wlan_data.bt_enable_gpio = GPIO_TO_PIN(0, 17);
+		am335xevm_wlan_data.bt_enable_gpio = GPIO_TO_PIN(3, 21);
 	}
 	else {
 		am335xevm_wlan_data.wlan_enable_gpio = GPIO_TO_PIN(1, 30);
diff --git a/arch/arm/mach-omap2/mux33xx.c b/arch/arm/mach-omap2/mux33xx.c
index 8e9180f..a7e120f 100644
--- a/arch/arm/mach-omap2/mux33xx.c
+++ b/arch/arm/mach-omap2/mux33xx.c
@@ -355,7 +355,7 @@ static struct omap_mux __initdata am33xx_muxmodes[] = {
 		NULL, NULL, NULL, NULL),
 	_AM33XX_MUXENTRY(MCASP0_AHCLKX, 0,
 		"mcasp0_ahclkx", "mcasp0_axr3",	NULL, "mcasp1_axr1",
-		NULL, NULL, NULL, "gpio0_17"),
+		NULL, NULL, NULL, "gpio3_21"),
 	_AM33XX_MUXENTRY(XDMA_EVENT_INTR0, 0,
 		"xdma_event_intr0", NULL, NULL, NULL,
 		"spi1_cs1", NULL, NULL, NULL),
-- 
1.7.11.2

